source/Calamari.Aws/Deployment/Conventions/UploadAwsS3Convention.cs

```diff
@@ -365,7 +366,7 @@ private async Task<bool> ShouldUpload(AmazonS3Client client, PutObjectRequest re
                     return true;
 
                 var metadata = await client.GetObjectMetadataAsync(request.BucketName, request.Key);
-                return !metadata.GetEtag().IsSameAsRequestMd5Digest(request);
+                return !metadata.GetEtag().IsSameAsRequestMd5Digest(request) || !S3ObjectExtensions.MetadataEq(request.ToCombinedMetadata(), metadata.ToCombinedMetadata());
```

Comment:
Would this read more clearly as `metadata.HasSameMetadata(request)` or similar (using the extension method)? I had to read the line several times to realise it's trying to express "has the file content changed, or has the metadata changed".

Reply:
Moved this into an extension method, so should read a bit better in the location where it is actually being used